lib/ansible/modules/pip.py

```diff
@@ -611,6 +613,61 @@ def setup_virtualenv(module, env, chdir, out, err):
     return out, err, cmd
 
 
+def _normalize_vcs_packages(module: AnsibleModule,
+                            package_list: list[Package],
+                            pip: list[str]) -> list[Package]:
+    """Return list[Package] where all vcs URL packages are converted to actual package names"""
+    vcs_packages = [str(pkg) for pkg in package_list if _is_vcs_url(str(pkg))]
```

Comment:
Would something like this avoid double-conversion?
```suggestion
    vcs_packages = [pkg_str for pkg in package_list if _is_vcs_url(pkg_str := str(pkg))]
```